python/orchestrator/orchestrator.py

```diff
@@ -0,0 +1,307 @@
+        # registry service manager
+        self.__component_service_registry_manager =\
+            component_service_registry_manager
+        # flag to indicate whether Orchestrator is registered with registry
+        self.__is_registered = multiprocessing.Event()
+        # instantiate global health and status manager object
+        self.__health_status_keeper = HealthStatusKeeper(
+                                    self._log_settings,
+                                    self._configurations_manager,
+                                    self.__component_service_registry_manager)
+        self.__alarm_signal_monitor = SignalMonitor(
+                                        self._log_settings,
+                                        self._configurations_manager,
+                                        self.__health_status_keeper,
+                                        self.__alarm_event)
+        self.__step_sizes = None
+        self.__response_codes = []
+        self.__command_and_steering_service = []
+        self.__command_and_steering_service_in_queue = None
+        self.__command_and_steering_service_out_queue = None
+        self.__orchestrator_registered_component = None
+        self.__communicator = None
+        self.__logger.debug("initialized.")
+
+    @property
+    def is_registered_in_registry(self): return self.__is_registered
+
+    def __get_component_from_registry(self,
+                                      target_components_category) -> list:
+        """
+        helper function for retreiving the proxy of
+        registered components by category.
+
+        Parameters
+        ----------
+        target_components_category : SERVICE_COMPONENT_CATEGORY.Enum
+            Category of target service components
+
+        Returns
+        ------
+        components: list
+            list of components which havecategory as target_components_category
+        """
+        components = self.__component_service_registry_manager.\
+            find_all_by_category(target_components_category)
+        self.__logger.debug(
+            f'found components: {len(components)}')
+        return components
+
+    def __update_local_state(self, registered_component, state):
+        """
+        helper function for updating the local state.
+
+         Parameters
+        ----------
+        registered_component : ServiceComponent
+            component registered in registry.
+
+        state : STATES.Enum
+            the new state of the component.
+
+        Returns
+        ------
+        response code: int
+            response code indicating whether or not the state is updated.
+        """
+        return self.__component_service_registry_manager.update_state(
+                                                        registered_component,
+                                                        state)
+
+    def __find_minimum_step_size(self, step_sizes_with_pids):
```

Comment:
Should this be a internal function with a single _ at the start?

Reply:
We make it private to Orchestrator to determine the minimum step_size after receiving the local step_sizes from all Application Companions. Then, Orchestrator will send this (global) minimum step_size to all Application Companions. However, if needed we can expose this minimum step_size through a public attribute.